main.go

```diff
@@ -180,29 +183,49 @@ func main() {
 				s.Suffix = fmt.Sprintf(" Deactivating the %v daemon", BrandName)
 				s.Start()
 				if err := deactivate(); err != nil {
-					return cli.Exit(err, 1)
+					errorMessages[BrandName] = fmt.Errorf("cannot deactivate daemon: %w", err)
+					s.Stop()
+					fmt.Printf(errorPrefix+" Cannot deactivate the %v daemon\n", BrandName)
+				} else {
+					s.Stop()
+					fmt.Printf(failPrefix+" Deactivated the %v daemon\n", BrandName)
 				}
-				s.Stop()
-				fmt.Printf(failPrefix+" Deactivated the %v daemon\n", BrandName)
 
 				s.Suffix = " Disconnecting from Red Hat Insights..."
 				s.Start()
 				if err := unregisterInsights(); err != nil {
-					return cli.Exit(err, 1)
+					errorMessages["insights"] = fmt.Errorf("cannot disconnect from Red Hat Insights: %w", err)
+					s.Stop()
+					fmt.Printf(errorPrefix + " Cannot disconnect from Red Hat Insights\n")
+				} else {
+					s.Stop()
+					fmt.Print(failPrefix + " Disconnected from Red Hat Insights\n")
 				}
-				s.Stop()
-				fmt.Print(failPrefix + " Disconnected from Red Hat Insights\n")
 
 				s.Suffix = " Disconnecting from Red Hat Subscription Management..."
 				s.Start()
 				if err := unregister(); err != nil {
-					return cli.Exit(err, 1)
+					errorMessages["rhsm"] = fmt.Errorf("cannot disconnect from Red Hat Subscription Management: %w", err)
+					s.Stop()
+					fmt.Printf(failPrefix + " Cannot disconnect from Red Hat Subscription Management\n")
```

Comment:
Here it should be:

`fmt.Printf(errorPrefix + " Cannot disconnect from Red Hat Subscription Management\n")`

Reply:
Good catch. Thank you!